src/instructlab/eval/ragas.py

```diff
@@ -0,0 +1,80 @@
+# Standard
+from typing import List, TypedDict
+
+# Third Party
+from langchain_community.chat_models import ChatOpenAI
+from ragas.evaluation import EvaluationDataset, EvaluationResult, RunConfig, evaluate
+from ragas.metrics import RubricsScore
+from ragas.metrics._domain_specific_rubrics import DEFAULT_WITH_REFERENCE_RUBRICS
+
+# Local
+from .evaluator import Evaluator
+
+
+class Sample(TypedDict):
+    # question
+    user_input: str
+
+    # model answer
+    response: str
+
+    # golden answer
+    reference: str
+
+
+class RagasEvaluator(Evaluator):
+    # most basic implementation, we just assume that the user will bring the existing model responses
+    name = "ragas"
+
+    def __init__(self):
+        pass
+
+    def run(
```

Comment:
So for this a user is expected to bring a list of Sample objects, which hold the input, prediction, and ground truth? Are we going to provide a way to build this list of Samples from given files or lists of each category, or is this moreso just for use with self-built scripts that import the Sample object and build?

Reply:
Updated it such that `dataset` now is either a `pathlib.Path` object or a `list` of samples, and we read what we need to accordingly